django/db/models/query.py

```diff
@@ -1203,21 +1204,22 @@ def _batched_insert(self, objs, fields, batch_size, ignore_conflicts=False):
             raise NotSupportedError('This database backend does not support ignoring conflicts.')
         ops = connections[self.db].ops
         batch_size = (batch_size or max(ops.bulk_batch_size(fields, objs), 1))
-        inserted_ids = []
+        inserted_rows = []
         bulk_return = connections[self.db].features.can_return_rows_from_bulk_insert
         for item in [objs[i:i + batch_size] for i in range(0, len(objs), batch_size)]:
             if bulk_return and not ignore_conflicts:
-                inserted_id = self._insert(
-                    item, fields=fields, using=self.db, return_id=True,
+                inserted_columns = self._insert(
+                    item, fields=fields, using=self.db,
+                    returning_fields=self.model._meta.db_returning_fields,
                     ignore_conflicts=ignore_conflicts,
                 )
-                if isinstance(inserted_id, list):
-                    inserted_ids.extend(inserted_id)
+                if isinstance(inserted_columns, list):
```

Comment:
Can we always assume we are getting a list returned? I believe `SQLInsertCompiler.execute_sql()` always returns a list now? (Although we may need a note in the backward incompatibility section for database backends?)

If all that were the case, we could possibly condense this whole section to something like the following:
```python
returning_fields = None
bulk_return = connections[self.db].features.can_return_rows_from_bulk_insert
if bulk_return and not ignore_conflicts:
    returning_fields = self.model._meta.db_returning_fields
return list(itertools.chain.from_iterable(
    self._insert(
        item, fields=fields, using=self.db,
        returning_fields=returning_fields,
        ignore_conflicts=ignore_conflicts,
    )
    for item in (objs[i:i + batch_size] for i in range(0, len(objs), batch_size)):
))
```

Reply:
Sadly that does not work, since execute does always return a list, but it can be a list of tuples. Or a list of a list of a tuple.